### PR TITLE
Make e2e test with local server run in parallel

### PR DESCRIPTION
**Pull request by curtisman**
Also fix a test in `deRehydrateContainerTests.spec.ts` that register the wrong `deltaManager` for the `OpProcessingController`.

### CODE REVIEW

File: packages/test/end-to-end-tests/package.json [hunk 1]
```diff
@@ -26,10 +26,11 @@
     "start:tinylicious:test": "tinylicious > tinylicious.log 2>&1",
     "test": "npm run test:realsvc",
     "test:realsvc": "npm run test:realsvc:local && npm run test:realsvc:tinylicious",
-    "test:realsvc:local": "mocha dist/test --config src/test/.mocharc.js",
+    "test:realsvc:local": "cross-env FLUID_TEST_DRIVER= npm run test:realsvc:run -- --parallel",
```
Comment (tylerbutler) [inline packages/test/end-to-end-tests/package.json:29]: Missed setting the environment variable?

**Review by tylerbutler — APPROVED** [review]

File: packages/test/end-to-end-tests/package.json (hunk 1 above)
Comment (curtisman) [reply packages/test/end-to-end-tests/package.json:29]: It clears it on purpose.

Comment (curtisman) [reply packages/test/end-to-end-tests/package.json:29]: Although I think we accept "local" too.... so that might be less confusing.  I will change it.

**Comment by curtisman** [comment]
Unfortunately, mocha-junit-reporter isn't compatible with --parallel yet.
https://github.com/michaelleeallen/mocha-junit-reporter/issues/140

**Comment by curtisman** [comment]
@tylerbutler just FYI, I have changed it to only enable if dev box for now.

File: packages/test/end-to-end-tests/src/test/.mocharc.js [hunk 2]
```diff
@@ -37,6 +38,11 @@ if(process.env.FLUID_TEST_REPORT === "1"){
     // give the report file a unique name based on driver config
     `mochaFile=${packageDir}/nyc/${testDriver}-junit-report.xml`
   ];
+
+  // mocha-junit-reporter doesn't support --parallel yet
+  // https://github.com/michaelleeallen/mocha-junit-reporter/issues/140
+  // That means the CI doesn't get the speed up.
+  config["parallel"] = false;
```
Comment (markfields) [inline packages/test/end-to-end-tests/src/test/.mocharc.js:45]: @curtisman so to get parallel mode you have to manually comment this out...?

**Pull request merged**